packages/vector_graphics/lib/src/_http_io.dart

```diff
@@ -0,0 +1,21 @@
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:flutter/foundation.dart';
+
+/// Fetches an HTTP resource from the specified [uri] using the specified [headers].
+Future<Uint8List> httpGet(Uri uri, {Map<String, String>? headers}) async {
+  final HttpClient httpClient = HttpClient();
+  final HttpClientRequest request = await httpClient.getUrl(uri);
+  if (headers != null) {
+    headers.forEach((String key, String value) {
+      request.headers.add(key, value);
+    });
+  }
+  final HttpClientResponse response = await request.close();
+
+  if (response.statusCode != HttpStatus.ok) {
+    throw HttpException('Could not get network asset', uri: uri);
```

Comment:
You need to drain the stream here - see the equivalent code in flutter/flutter

Reply:
does it really matter if we're going to throw though?

Reply:
Uhhh ... ask Zach :)

Reply:
I see now